Stable Vite and Vitest (#1424)
seek-oss/sku: docs/docs/configuration.md

Config files can use either TypeScript or JavaScript.

## bundler

> 
> Type: `string`
> 
> Default: `webpack`
> 
> The bundler that sku uses to build the application.
> 

Reviewer: I feel the type for this one should specify that this value only accepts a union `webpack` or `vite`

May also be worth adding a link to the Vite docs in here

```suggestion

Type: `webpack` | `vite`

Default: `webpack`

The bundler that sku uses to build the application.
See [Vite support](https://seek-oss.github.io/sku/#/./docs/vite) for details.
```

Or something along those lines

Config files can use either TypeScript or JavaScript.

## bundler

Type: `webpack | vite`

Default: `webpack`

The bundler that sku uses to build the application.
See [Vite support](https://seek-oss.github.io/sku/#/./docs/vite) for details.

> 

Reviewer 2: Does this need some mention of "vite is only supported for static apps currently"?

Author: Yea good call. I found a few other experimental notices to remove too

```

## dangerouslySetViteConfig

Type: `function`

Bundler: `vite`

This function provides a way to modify sku's Vite configuration.
It should only be used in exceptional circumstances where a solution cannot be achieved by adjusting standard configuration options.

Before customizing your Vite configuration, please reach out in [#sku-support](https://seek.enterprise.slack.com/archives/CDL5VP5NU) to discuss your requirements and potential alternative solutions.

As sku creates two Vite configs (`client` & `render`), this function will actually run twice.
If you only need to modify one of these configs, then you can check `env.mode` from the second argument within.

This function can return a partial config object that will be deeply merged into existing config (recommended), or directly mutate the config (if the default merging cannot achieve the desired result).

> Sku provides no guarantees that its Vite configuration will remain compatible with any customizations made within this function.
> It is the responsibility of the user to ensure that their customizations are compatible with sku.

Example:

```ts
export default {
  // partial config is deeply merged

>   dangerouslySetViteConfig: () => ({

Reviewer: It may be beneficial to list what args are provided here

```suggestion
  dangerouslySetViteConfig: (config, env) => ({ // or whatever the actual args are
```

```typescript
export default {

>   bundler: 'vite',

Reviewer: Our little boy is all grown up 🥹

The directory to build your assets into when running `sku build` or `sku build-ssr`

## testRunner

> Type: `string`

Reviewer: Similar to the bundler one, I think this should specify that its a union

```suggestion
Type: `jest` | `vitest`
```

That would allow us to not need to specify the valid options below

This is used to enable CommonJS interop for these dependencies when using the `vite` bundler.

> See https://github.com/cyco130/vite-plugin-cjs-interop for more information.

Author: cjsInteropDependencies is still marked as unsafe since we are still unsure if we need this at all. It's available for potential issues that may arise during migration but the impact is uncertain. 

In future this will either be removed or made stable.
